### ISSUE TITLE
[Question] Catch exceptionCaught() event was fired, and it reached at the tail of the pipeline

### ISSUE THREAD

**Issue by MarcMagnin** [grpc/grpc-java#3020]
### What version of gRPC are you using?
1.2.0

### What JVM are you using (`java -version`)?
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

### What did you do?
Connect to a Unix domain socket that is not existing anymore with the following:
```java
NettyChannelBuilder.forAddress(new DomainSocketAddress("/tmp/socket123.so"))
                    .eventLoopGroup(new EpollEventLoopGroup())
                    .channelType(EpollDomainSocketChannel.class)
                    .usePlaintext(true)
                    .build();
```

Then those WARN will be produced in the output:
```
WARN  [epollEventLoopGroup-2-5] 2017-05-18 23:10:17,648  Slf4JLogger.java:151 - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. 
It usually means the last handler in the pipeline did not handle the exception.
test.io.netty.channel.AbstractChannel$AnnotatedConnectException: connect() failed: No such file or directory: /tmp/socket123.so
```

How can we handle those exceptions or make them silent?

Many thanks!

**Comment by ejona86** [comment]
This should be handled so that we can notify the RPC the cause of the failure. Some changes to this area happened in 1.3.0; it may handle it better. Some of the changes I've not had time to land for better proxy error handling may also fix this.

**Comment by ejona86** [comment]
The fix in 1.3.0 should have helped, but #3360 is the last of a series of fixes that improved error handling during early connection establishment and negotiation.
